packages/ra-auth-google/src/GoogleLoginButton.tsx

```diff
@@ -1,43 +1,43 @@
-import { Box, SxProps, styled } from "@mui/material";
-import * as React from "react";
-import { useLogin } from "react-admin";
-import { GsiButtonConfiguration, IdConfiguration } from "./types";
+import { Box, SxProps, styled } from '@mui/material';
+import * as React from 'react';
+import { useLogin } from 'react-admin';
+import { GsiButtonConfiguration, IdConfiguration } from './types';
 
-const GoogleButton = (props: Omit<GoogleLoginButtonProps, "sx">) => {
-  const { gsiParams, ...buttonProps } = props;
-  const login = useLogin();
-  const divRef = React.useRef<HTMLDivElement>(null);
+const GoogleButton = (props: Omit<GoogleLoginButtonProps, 'sx'>) => {
+    const { gsiParams, ...buttonProps } = props;
+    const login = useLogin();
+    const divRef = React.useRef<HTMLDivElement>(null);
 
-  React.useEffect(() => {
-    if (!window?.google || !divRef.current) {
-      return;
-    }
+    React.useEffect(() => {
+        if (!window?.google || !divRef.current) {
+            return;
+        }
 
-    window.google.accounts.id.initialize({
-      ...gsiParams,
-      callback: login,
-    });
+        window.google.accounts.id.initialize({
+            ...gsiParams,
+            callback: login,
+        });
 
-    window.google.accounts.id.renderButton(divRef.current, {
-      width: "300px",
-      ...buttonProps,
-    });
-  }, [window?.google, divRef.current]);
+        window.google.accounts.id.renderButton(divRef.current, {
+            width: '300px',
+            ...buttonProps,
+        });
+    }, [gsiParams, login, buttonProps]);
```

Comment:
Changing the dependencies of a useEffect is a major change! You should not trust the linter blindly. Adding the missing deps `gsiParams, login, buttonProps` was certainly a good idea, but removing `window?.google, divRef.current` was not. These were here for a reason.